packages/components/src/Combobox/ComboboxProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { RefObject, useRef, useState } from "react";
```

Comment:
Very much a fan of this pattern to avoid prop drilling, great work.

Naive, philosophical new person question: Do we provide any way of consumers to override this internal state? I can definitely see a future where consumers of the combobox component will want to open/close it some other way than a trigger button, or say, manually set the search field content because the user clicked on a specific button within a UI to represent a common search term. I'm spitballing on the use cases of overriding internal state (there's a hundred others), but I'm definitely a fan of allowing consumers to override (it increases extensibility without much effort on our behalf). That said, if we've made the collective decision to not allow override for specific reasons, I'm always open to having my mind changed especially for our use cases.

Reply:
hmm those are interesting scenarios. my question with those would be: is that a combobox we're talking about or maybe something else at that point?

I'm not totally opposed to opening things up to consumers, but we've gotta have some limits. I wouldn't want us to enable too much customization to the point where the behavior is quite different depending on how much of the default behavior has been overridden